.github/workflows/refresh-sandbox.yml

```diff
@@ -1,6 +1,9 @@
 name: "Refresh Salesforce Sandbox"
 
 on:
+  repository_dispatch:
+    types: [external-trigger-sandbox-refresh]
+
```

Comment:
Have you looked at workflow_call? https://docs.github.com/en/actions/writing-workflows/choosing-when-your-workflow-runs/events-that-trigger-workflows#workflow_call